### PR TITLE
Remove arguments from perform method

### PR DESCRIPTION
**Pull request by morgoth**
Follow up to https://github.com/bensheldon/good_job/issues/136#issuecomment-693451491

### CODE REVIEW

**Comment by morgoth** [comment]
The linter fails because code uses Ruby 2.5 syntax and Rubocop is set to lint against 2.4 (the same as in gemspec https://github.com/bensheldon/good_job/blob/main/good_job.gemspec#L48) but there are no tests for Ruby 2.4.
Should I change the code to 2.4 syntax or do you think it is fine to bump gem to use 2.5?

**Comment by bensheldon** [comment]
@morgoth Oof, sorry you got bitten by that inconsistency. Let's drop support for Ruby 2.4; it looks like [Ruby 2.4 EOLed on March 31, 2020](https://www.ruby-lang.org/en/news/2020/04/05/support-of-ruby-2-4-has-ended/). 

Could you please remove the references to 2.4, and change the Linter config to 2.5 as its own PR?

File: spec/lib/good_job/job_spec.rb [hunk 1]
```diff
@@ -204,11 +204,16 @@ def perform(result_value = nil, raise_error: false)
     end
 
     it 'can preserve the job' do
-      good_job.perform(destroy_after: false)
+      current_preserve_job_records = GoodJob.preserve_job_records
+      GoodJob.preserve_job_records = true
```
Comment (bensheldon) [inline spec/lib/good_job/job_spec.rb:208]: The RSpec-y way to do this is with a [message stub](https://relishapp.com/rspec/rspec-mocks/v/3-9/docs/configuring-responses) which will be automatically cleaned up and reset at the end of the test example:  `allow(GoodJob).to receive(:preserve_job_records).and_return(true)`

File: lib/good_job/job.rb [hunk 2]
```diff
@@ -215,15 +207,15 @@ def perform(destroy_after: !GoodJob.preserve_job_records, reperform_on_standard_
...
       else
         self.finished_at = Time.current
 
-        if destroy_after
-          destroy!
-        else
+        if GoodJob.preserve_job_records
           save!
+        else
+          destroy!
```
Comment (bensheldon) [inline lib/good_job/job.rb:218]: This reads so much better now with the positive condition coming first 🎉

**Review by bensheldon — COMMENTED** [review]
This is fantastic! 🙌 

I left some feedback on the tests, specifically to use RSpec message stubs instead of manually setting and resetting global values. I have to apologize (and express my gratitude to you for working in them) because those tests could be more exemplary.

Once you fix the linter issue, and make the changes to the tests, this is ready to go 👍

**Comment by bensheldon** [comment]
@morgoth I just merged #142 to unblock this.

**Comment by morgoth** [comment]
@bensheldon updated and tests are green now. Thanks

**Comment by bensheldon** [comment]
@morgoth Thank _you_! 🙌

**Pull request merged**